apm-sniffer/apm-sdk-plugin/spring-plugins/core-patch/src/main/java/org/skywalking/apm/plugin/spring/patch/CreateAopProxyInterceptor.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2017, OpenSkywalking Organization All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Project repository: https://github.com/OpenSkywalking/skywalking
+ */
+
+package org.skywalking.apm.plugin.spring.patch;
+
+import java.lang.reflect.Method;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.springframework.aop.framework.AdvisedSupport;
+
+/**
+ * {@link CreateAopProxyInterceptor} check that the bean has been implement {@link EnhancedInstance}. <p/>
+ * if yes, true will be returned.
+ *
+ * @author zhang xin
+ */
+public class CreateAopProxyInterceptor implements InstanceMethodsAroundInterceptor {
+    private static final String ENHANCE_INSTANCE_INTERFACE_NAME = EnhancedInstance.class.getName();
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+        MethodInterceptResult result) throws Throwable {
+
+    }
+
+    @Override
+    public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+        Object ret) throws Throwable {
+        AdvisedSupport advisedSupport = (AdvisedSupport)allArguments[0];
+        Class[] interfaces = advisedSupport.getTargetClass().getInterfaces();
```

Comment:
You should use `Class#isAssignableFrom` to make sure this class implement the `EnhancedInstance` interface.